providers/core.tcpsocket.js

```diff
@@ -359,17 +360,24 @@ TcpSocket_node.prototype.onAccept = function (connection) {
  * @param {Function} continuation Function to call once socket is disconnected.
  */
 TcpSocket_node.prototype.close = function (continuation) {
-  if (this.connection) {
-    if (this.state === TcpSocket_node.state.BINDING ||
-        this.state === TcpSocket_node.state.LISTENING) {
-      this.connection.close(continuation);
-    } else {
+  if (this.connection && this.state !== TcpSocket_node.state.CLOSED) {
+    try {
+      if (this.state === TcpSocket_node.state.LISTENING) {
+        // Close server socket
+        this.connection.close();
+      } else {
+        // Close client socket
+        this.connection.end();
+      }
+    } catch(e) {
+      // Definitely close one way or another
+      console.warn("Had to destroy socket " + this.id);
       this.connection.destroy();
-      continuation();
     }
     delete this.connection;
     this.state = TcpSocket_node.state.CLOSED;
     TcpSocket_node.connectionState[this.id] = this.state;
+    continuation();
```

Comment:
I don't think so - at least, by marking the state closed this branch shouldn't be repeated (further calls should just end in the error already closed branch). But maybe I'm missing your question?